src/main/java/com/xpinjection/springboot/dao/valueobject/Recommendation.java

```diff
@@ -0,0 +1,12 @@
+package com.xpinjection.springboot.dao.valueobject;
+
+import lombok.Value;
+
+@Value
+public class Recommendation {
+    private final String instantiator;
+
+    public String toString(){
```

Comment:
No need to override toString method, it is generated from Value annotation.

Reply:
I did it purpose to avoid Lombok's automatic flooding 
`        return "Recommendation(instantiator=" + this.getInstantiator() + ")";`
so it behaves just like your code in the current state of the project